### DISCUSSION TITLE
Ideas for contributions 💡

### DISCUSSION THREAD

**Discussion by dargmuesli** [spotify-web-api-java/spotify-web-api-java#385] [Announcements]
Hey there! This project provides a pretty solid wrapper for the Spotify Web API. Although stable, there are some things that would benefit from you contributing! :rocket: 

- [ ] migrate to Gradle: it seems that for some good reasons most modern projects use Gradle
- [ ] migrate to Kotlin: the hypothesis to be proven here is a reduction in boilerplate code
- [x] general code cleanup: IDEs like IntelliJ IDEA suggest some various improvements across the codebase which should be applied
- [ ] code formatting tooling: there should be some linting to ensure consistent code formatting in all relevant files

Pick one, communicate your pick and open a PR! About a thousand stargazers would very much appreciate your work :tada:

**Comment by Selbi182** [comment]
I mean, I could do the IntelliJ IDE cleanup. Just running _Inspect project_ for any warnings and stuff shouldn't be too difficult.

**Reply by dargmuesli** [reply]
That would be nice, I really like to open projects in an IDE and not see yellow squiggly lines :grin:

**Reply by Selbi182** [reply]
Alright. I'll try to get to get it done by the end of the week

**Reply by dargmuesli** [reply]
Ok, no hurry, it's just some beautification 🎨 if anything's more urgent, do that first, it's a nice to have :wink:

**Reply by Selbi182** [reply]
![grafik](https://github.com/spotify-web-api-java/spotify-web-api-java/assets/8850085/c7ea7b42-fad2-4424-95e5-9d0330fb8ac9)
This might take a while...

**Reply by dargmuesli** [reply]
Those HTML / JS / TS warnings can most likely be ignored as they should occur in the automatically generated documentation. Of course it would be cool if some IDEA configuration could be added so that generated code is excluded from inspections.

**Reply by Selbi182** [reply]
https://github.com/spotify-web-api-java/spotify-web-api-java/pull/386 PR is up, though I'd ask for a second opinion. This was pretty underwhelming lol

**Reply by alerGeek** [reply]
Hey @dargmuesli ,
Maybe I will take a look at one of topic next week. Regarding the migration to gradle/kotlin - do I correctly understand, that it is about replacing maven way of building application. In this case make sense to only one of these options should be chosen?

Or are we talking about:
1) migrating software automation system from maven to gradle
2) (and) migrating application langues from java to kotlin?

**Comment by dargmuesli** [comment]
@alerGeek let's open a new comment thead for your question.
Yes, migration to Gradle should be separate from migrating the programming language.

**Reply by dargmuesli** [reply]
The main question that's needed to be answered is if Gradle has a working replacement for `maven-gpg-plugin`. The gpg signing has been setup by the previous maintainer for this repository and I'm not 100% sure if a simple migration is possible, but it should be.
